quickwit/quickwit-ingest/src/ingest_v2/workbench.rs

```diff
@@ -137,44 +134,39 @@ impl IngestWorkbench {
         subworkbench.persist_success_opt = Some(persist_success);
     }
 
-    pub fn record_persist_failure(&mut self, persist_failure: &PersistFailure) {
-        let Some(subworkbench) = self.subworkbenches.get_mut(&persist_failure.subrequest_id) else {
-            warn!(
-                "could not find subrequest `{}` in workbench",
-                persist_failure.subrequest_id
-            );
+    fn record_failure(&mut self, subrequest_id: SubrequestId, failure: SubworkbenchFailure) {
+        let Some(subworkbench) = self.subworkbenches.get_mut(&subrequest_id) else {
+            warn!("could not find subrequest `{}` in workbench", subrequest_id);
             return;
         };
         subworkbench.num_attempts += 1;
-        subworkbench.last_failure_opt = Some(SubworkbenchFailure::Persist((
-            persist_failure.shard_id,
-            persist_failure.reason(),
-        )));
+        subworkbench.last_failure_opt = Some(failure);
     }
 
     pub fn record_no_shards_available(&mut self, subrequest_id: SubrequestId) {
-        let Some(subworkbench) = self.subworkbenches.get_mut(&subrequest_id) else {
-            warn!("could not find subrequest `{}` in workbench", subrequest_id);
-            return;
-        };
-        subworkbench.num_attempts += 1;
-        subworkbench.last_failure_opt = Some(SubworkbenchFailure::NoShardsAvailable);
+        self.record_failure(subrequest_id, SubworkbenchFailure::NoShardsAvailable);
+    }
+
+    /// Marks a node as unavailable for the span of the workbench.
+    ///
+    /// Remaining attempts will treat the node as if it was not in the ingester pool.
+    pub fn record_connection_error(&mut self, subrequest_id: SubrequestId) {
+        self.record_failure(subrequest_id, SubworkbenchFailure::ConnectionError);
```

Comment:
Is a connection error semantically different than a transport error, and do we need to distinguish between them? If not, can we use the same name?